### PR TITLE
Avoid duplicate meta data rows in import_spss()

### PR DESCRIPTION
**Pull request by beckerbenj**
In `SPSS`, character variables can have arbitrary value labels and missing tags. For instance, both `99` and `99.0` can be labelled. However, `eatGADS` transforms all labelled (and missing tagged) values to numeric, resulting in duplicate value labels for `99`. Such duplicate meta data rows result in errors in, e.g., `recodeGADS()`, where duplicate meta data is checked explicitly.

This PR adds functionality that removes such duplicate value rows, addressing #51

### CODE REVIEW

**Comment by beckerbenj** [comment]
It should be added:
`prepare_labels()` could benefit from refactoring, as its architecture is more complex than it probably needs to be and performance could be enhanced by reducing redundancies. However, this is well beyond the scope of the current PR.

**Review by franikowsp — APPROVED** [review]
LGTM! Only tiny nits.

File: R/extract_labels.R [hunk 1]
```diff
@@ -101,7 +101,7 @@ extract_value_level.haven_labelled <- function(var, varName) {
     valLabels <- character()
   }
 
-  # transform values to numeric if possible, leave characters as is (rest is taking care of in char_valLabels2numeric.R)
+  # transform values to numeric if possible, leave characters as is (rest has been taking care of in char_valLabels2numeric.R)
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply R/extract_labels.R:104]: Agree and corrected

File: tests/testthat/test_extract_labels.R [hunk 2]
```diff
@@ -54,6 +54,29 @@ test_that("Value label of single variable extracted for SPSS types", {
                data.frame(varName = "VAR1", value = 1, valLabel = "One", missings = "valid", stringsAsFactors = FALSE))
 })
 
+test_that("Conflicting value label for string variable from SPSS", {
+  f <- haven::labelled("a",
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply tests/testthat/test_extract_labels.R:58]: Agree and modified

File: R/extract_labels.R [hunk 3]
```diff
@@ -111,13 +111,33 @@ extract_value_level.haven_labelled <- function(var, varName) {
...
 
+remove_duplicate_value_rows <- function(df, varName) {
+  values_with_dups <- unique(df$value[duplicated(df$value)])
+  # tbd: if all equal, select first row
+  # otherwise select first row and raise warning? should this be parameterized via an argument?
+  if(length(values_with_dups) > 0) {
+    warning("Duplicate value labels or missing tags are dropped for variable '", varName,"'.")
+    for(i in values_with_dups){
+      dup_rows <- which(df$value == i)
+      rows_to_remove <- dup_rows[!dup_rows == min(dup_rows)]
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply R/extract_labels.R:134]: Agree and modified 😆

File: R/extract_labels.R [hunk 4]
```diff
@@ -111,13 +111,33 @@ extract_value_level.haven_labelled <- function(var, varName) {
...
   rownames(df) <- NULL
   df
 }
 
+remove_duplicate_value_rows <- function(df, varName) {
+  values_with_dups <- unique(df$value[duplicated(df$value)])
+  # tbd: if all equal, select first row
+  # otherwise select first row and raise warning? should this be parameterized via an argument?
+  if(length(values_with_dups) > 0) {
+    warning("Duplicate value labels or missing tags are dropped for variable '", varName,"'.")
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply R/extract_labels.R:131]: Agree and modified into:
```
warning("Duplicate value labels or missing tags are dropped for variable '", varName,
         "'. Only the respective first value label or missing tag is preserved and zero-decimals are dropped.")
```

**Pull request merged**